src/modules/StopLimitOrders.ts

```diff
@@ -0,0 +1,594 @@
+    order: SignedStopLimitOrder,
+  ): string {
+    const signatureBytes = Web3.utils.hexToBytes(order.typedSignature);
+    const byteArray = this.orderToByteArray(order).concat(signatureBytes);
+    return Web3.utils.bytesToHex(byteArray);
+  }
+
+  // ============ Private Helper Functions ============
+
+  private orderToByteArray(
+    order: StopLimitOrder,
+  ): number[] {
+    return []
+      .concat(argToBytes(order.makerMarket))
+      .concat(argToBytes(order.takerMarket))
+      .concat(argToBytes(order.makerAmount))
+      .concat(argToBytes(order.takerAmount))
+      .concat(argToBytes(order.makerAccountOwner))
+      .concat(argToBytes(order.makerAccountNumber))
+      .concat(argToBytes(order.takerAccountOwner))
+      .concat(argToBytes(order.takerAccountNumber))
+      .concat(argToBytes(order.triggerPrice))
+      .concat(argToBytes(order.decreaseOnly))
+      .concat(argToBytes(order.expiration))
+      .concat(argToBytes(order.salt));
+  }
+
+  private getDomainData() {
+    return {
+      name: 'StopLimitOrders',
+      version: '1.0',
+      chainId: this.networkId,
+      verifyingContract: this.contracts.stopLimitOrders.options.address,
+    };
+  }
+
+  private async ethSignTypedOrderInternal(
+    order: StopLimitOrder,
+    signingMethod: SigningMethod,
+  ): Promise<string> {
+    const orderData = {
+      makerMarket: order.makerMarket.toFixed(0),
+      takerMarket: order.takerMarket.toFixed(0),
+      makerAmount: order.makerAmount.toFixed(0),
+      takerAmount: order.takerAmount.toFixed(0),
+      makerAccountOwner: order.makerAccountOwner,
+      makerAccountNumber: order.makerAccountNumber.toFixed(0),
+      takerAccountOwner: order.takerAccountOwner,
+      takerAccountNumber: order.takerAccountNumber.toFixed(0),
+      triggerPrice: order.triggerPrice.toFixed(0),
+      decreaseOnly: order.decreaseOnly,
+      expiration: order.expiration.toFixed(0),
+      salt: order.salt.toFixed(0),
+    };
+    const data = {
+      types: {
+        EIP712Domain: EIP712_DOMAIN_STRUCT,
+        StopLimitOrder: EIP712_ORDER_STRUCT,
+      },
+      domain: this.getDomainData(),
+      primaryType: 'StopLimitOrder',
+      message: orderData,
+    };
+    return this.ethSignTypedDataInternal(
+      order.makerAccountOwner,
+      data,
+      signingMethod,
+    );
+  }
+
+  private async ethSignTypedCancelOrderInternal(
+    orderHash: string,
+    signer: string,
+    signingMethod: SigningMethod,
+  ): Promise<string> {
+    const data = {
+      types: {
+        EIP712Domain: EIP712_DOMAIN_STRUCT,
+        CancelLimitOrder: EIP712_CANCEL_ORDER_STRUCT,
+      },
+      domain: this.getDomainData(),
+      primaryType: 'CancelLimitOrder',
+      message: {
+        action: 'Cancel Orders',
+        orderHashes: [orderHash],
+      },
+    };
+    return this.ethSignTypedDataInternal(
+      signer,
+      data,
+      signingMethod,
+    );
+  }
+
+  private async ethSignTypedDataInternal(
```

Comment:
did you just copy this from the limit orders one? Any way to share the code?

Reply:
changed